rules/dead-code/src/Rector/MethodCall/RemoveEmptyMethodCallRector.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DeadCode\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Nop;
+use PhpParser\Parser;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Type\ObjectType;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\DeadCode\Tests\Rector\MethodCall\RemoveEmptyMethodCallRector\RemoveEmptyMethodCallRectorTest
+ */
+final class RemoveEmptyMethodCallRector extends AbstractRector
+{
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    public function __construct(Parser $parser)
+    {
+        $this->parser = $parser;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Remove empty method call', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function callThis()
+    {
+    }
+}
+
+$some = new SomeClass();
+$some->callThis();
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function callThis()
+    {
+    }
+}
+
+$some = new SomeClass();
+$some->callThis();
```

Comment:
I think this line should be removed, right?

Btw, we could make a PHPStan rule fo this, so before/after strings in `new CodeSample()` are always different here. What do you think?

Reply:
Fixed. Yes, it seems we can create PHPStan rule for it.

Reply:
Could you make an issue for PHPStan so we don't forget?

Reply:
Yes https://github.com/rectorphp/rector/issues/4433

Reply:
Thank you